corehq/apps/notifications/models.py

```diff
@@ -58,7 +61,35 @@ def _fmt_note(note_idx):
             }
             return note_dict
 
-        return list(map(_fmt_note, enumerate(notes)))
+        notifications = list(map(_fmt_note, enumerate(notes)))
+
+        # pushes feature discovery notification to new web users
+        if not notifications:
+            all_notes = cls.objects.filter(Q(domain_specific=False) | Q(domains__overlap=couch_user.domains),
+                                        is_active=True)[:limit]
```

Comment:
Right, if a new user joins after the feature discovery notification was activated, they'll never see it, so if no notification was retrieved, all past notifications are retrieved and only the feature discovery ones are kept. This happens before it gets potentially filtered out due to the domain they're look at it from. This wasn't outlined in the ticket but I think Kirti and I talked about that scenario and decided that they should also see it.

Reply:
Ah sure that makes sense. Thanks for the explanation. In that case, I think it makes sense to keep that logic in this method since that is still dealing with obtaining the notifications in the first place. So move my "from here down" comment to below this 😅